### PR TITLE
chore: improve testing and cpu bound concurrency

### PR DESCRIPTION
**Pull request by danlucian**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

- Switch from hard coded concurrency level to CPU bound level
- Added ‘Critical’ severity type into the testing flows/fixtures
- Added extra tests to cover cases like: --print-deps/paths args, errors being thrown within the display call chain or the debug flag
- Renamed some functions reflecting their truly behaviour
- Renamed some functions reflecting their truly behaviour
- Added validation to include only non-empty files for hashing

**Before**:
before

**After**:
after

### CODE REVIEW

**Review by anthogez — COMMENTED** [review]
This should be a feat instead of a chore.

File: package.json [hunk 1]
```diff
@@ -29,26 +29,26 @@
     "prepare": "npm run build"
   },
   "dependencies": {
-    "@snyk/dep-graph": "^1.19.3",
-    "chalk": "^4.1.0",
-    "debug": "^4.1.1",
-    "hosted-git-info": "^3.0.7",
+    "@snyk/dep-graph": "^1.30.0",
```
Comment (anthogez) [inline package.json:32]: Would suggest this to be splitted in multiple pull requests as we are doing several changes..

**Comment by danlucian** [comment]
> This should be a feat instead of a chore.

I don't see it as a feature since we don't add anything new. The changes are about code cleanup, some extra tests and few lib updates (without impacting the already used APIs). What do you think?

File: package.json (hunk 1 above)
Comment (dekelund) [reply package.json:32]: If it's possible, then I might agree with @anthogez.

It's easier to trackback potential problems if we are able to checkout different changes when debugging. In this case we upgrade the dep-graph at the same time as we introduce new conditions in the find-algorithm (which shouldn't be a problem, but you never know)

Comment (danlucian) [reply package.json:32]: Okey dokey. We will have another PR for the versions.

**Comment by dekelund** [comment]
Looks similar to the implementation from our session, except the things we discussed yesterday, that was partly reverted.

**Review by dekelund — APPROVED** [review]
Approved, but I agree with Antonio that it would be a great thing to split the PR, so that we bump the version in another commit.

(Maybe a split for changes to find, and the changes to concurrency.)

**Comment by anthogez** [comment]
> > This should be a feat instead of a chore.
> 
> I don't see it as a feature since we don't add anything new. The changes are about code cleanup, some extra tests and few lib updates (without impacting the already used APIs). What do you think?

Concurrency based on the CPU would be the feat on this case, as we would no longer base our flows in a static number but in a new capability, in this case it’s welcomed as an improvement.

**Comment by danlucian** [comment]
Oh, okay, got it! I was thinking about a feature more as a product capability, not an inner implementation. But you are right! Thank you! 🤗

**Pull request merged**